packages/eas-cli/src/build/runBuildAndSubmit.ts

```diff
@@ -587,3 +608,56 @@ async function validateExpoUpdatesInstalledAsProjectDependencyAsync({
     }
   }
 }
+
+function isEnvironment(env: string): env is EnvironmentVariableEnvironment {
+  if (
+    Object.values(EnvironmentVariableEnvironment).includes(env as EnvironmentVariableEnvironment)
+  ) {
+    return true;
+  }
+  return false;
+}
+
+async function resolveEnvVarsAsync({
+  sessionManager,
+  flags,
+  buildProfile,
+  graphqlClient,
+  projectId,
+}: {
+  sessionManager: SessionManager | undefined;
+  flags: BuildFlags;
+  buildProfile: ProfileData<'build'>;
+  graphqlClient: ExpoGraphqlClient;
+  projectId: string;
+}): Promise<Record<string, string>> {
+  const environment = flags.environment ?? buildProfile.profile.environment;
+
+  if (!environment || !isEnvironment(environment)) {
+    return {};
```

Comment:
Shouldn't this be
```suggestion
    return {...profileEnv};
```
?

Reply:
That's a good question. I want the behaviour of local builds to be unchanged if the new `--environment` option is not used. Currently envs from `buildProfile` are passed to the runner via `job` object, and this behaviour is kept. When `--environment` is passed, envvars from `eas` and `buildProfile` are merged with preference for `eas` vars. Now that I think about it we can probably remove buildProfile vars from here entirely, since they are used in build anyway.